go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/gruntwork-io/terratest
 
-go 1.19
+go 1.21.1
```

Comment:
that is quite unusual to have a _patch_ version requirement in the go.mod (`go 1.21.1` instead of `go 1.21`).
it also means that all projects that import `github.com/gruntwork-io/terratest` will need to have the same requirement, and their `go.mod` file will be mutated when running `go mod tidy`.
is there a specific reason for the strict requirement?

Reply:
Hi, as far as I understand, the strict requirement is done by golang since the 1.21 version. Even if the go.mod file only has `1.21`, it implies that all the modules that depend on this need to have at least go 1.21.0 installed.
https://tip.golang.org/doc/toolchain#config

In regards to the add of the patch version, since 1.21 the go toolchain requires the patch version as part of the name, so to avoid issues is recommended to add the patch as part of the go.mod version
https://github.com/golang/go/issues/62278

Reply:
Thanks for the link. I was not aware of this change in go 1.21. But based on the discussion there, it feels like the go directive here should be `go 1.21.0`, and not `go 1.21.1`, which forces all modules using this one to also use `go 1.21.1`.

Reply:
I chose 1.21.1 because it was the latest version at that moment, but sure, I'll submit a PR to downgrade the version to 1.21.0.

Reply:
PR: https://github.com/gruntwork-io/terratest/pull/1349